### DISCUSSION TITLE
Allow re-editing of issue (esp. Description) if error occurs on creation

### DISCUSSION THREAD

**Discussion by bobobox** [ankitpokhrel/jira-cli#490] [Ideas]
I've often encountered a frustrating situation where I create a new issue, spend a lot of time writing a good, detailed Description, and then Submit, only to have some error cause everything to be discarded. All my work is gone.

It would be great if an error on submission would just bump me back to a state where I could correct the issues rather than lose everything.

Thanks!
R

**Comment by ankitpokhrel** [comment]
Hi @bobobox, I've faced this problem couple of times and I know it is frustrating. This is something on my list but I can't give you a timeline yet. The idea is to store all inputs temporarily as a json before sending the request and allow to use that if request fails. Github CLI does something similar.

This may not be ideal but as a temporary workaround, you can write detailed description in a text file and pass it as a template.

```sh
# Load description from template file
$ jira issue create --template /path/to/template.tmpl
```

**Reply by snpefk** [reply]
I think that for MVP that's gonna be enough. Logs can be stored in `/tmp/jira` or `/var/log/jira`. It's ok solution for power user, but not user friendly as it could be. Maybe on I/O error `jira-cli` should print error message with note pointing where the data can be recovered from? Also, json seems fine, but if `jira-cli` will log plain text with description user can retry by specifying log file with `--template` flag.

**Reply by ankitpokhrel** [reply]
> Maybe on I/O error jira-cli should print error message with note pointing where the data can be recovered from?

Yes, it would display something like this

```sh
✗ Operation failed. Use the following command to restore.
jira issue create --recover /var/log/jira/ds0sl0y-1670668698.json
```

And the logged JSON will have contents as below
```sh
{
  "project": "KEY",
  "type": "Task",
  "summary": "Title of the ticket",
  "body": "## Title\n\nSome description\n",
  "priority": "High",
  "assignee": "",
  "labels": ["p1", "q1"],
  "components": ["BE", "CLI"]
}

```

**Comment by martinpovolny** [comment]
Let's discuss how the feature should work. I can do it.
Meanwhile here's a patch that saves my day when JIRA server fails (only `jira issue create ....`)

```
diff --git a/internal/cmd/issue/create/create.go b/internal/cmd/issue/create/create.go
index cf3427b..c729c50 100644
--- a/internal/cmd/issue/create/create.go
+++ b/internal/cmd/issue/create/create.go
@@ -152,6 +152,7 @@ func create(cmd *cobra.Command, _ []string) {

                resp, err := client.CreateV2(&cr)
                if err != nil {
+                       fmt.Printf("\nCreation failed. Text dump follows.\nSummary:\n--------\n%s\n\nBody:\n-----\n%s\n\n", params.summary, params.body)
                        return "", err
                }
                return resp.Key, nil
```

**Comment by martinpovolny** [comment]
From my perspective, the only value is the `body`, all other data I have on the command line.
That is for me, it's enough to print the body text.

 * Either to stdout/stderr or 
 * to a file and print the file name (that could be used with the --template flag).

**Reply by ankitpokhrel** [reply]
It depends on the usecase I think. For my personal use I would also mostly care about the body. But it would be nice to be able to recover the entire command as is, some setup has custom fields that accepts long texts for example.

**Reply by martinpovolny** [reply]
@ankitpokhrel : shall I convert the above patch to a PR so that we have at least the body before someone comes up with a more complex solution?
